pkgs/by-name/fu/fusion/package.nix

```diff
@@ -0,0 +1,75 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  buildNpmPackage,
+  buildGoModule,
+  mockgen,
+  nixosTests,
+  nix-update-script,
+}:
+buildGoModule rec {
+  pname = "fusion";
+  version = "0.8.9";
+
+  src = fetchFromGitHub {
+    owner = "0x2E";
+    repo = "fusion";
+    rev = "v${version}";
+    hash = "sha256-nI587lshHlwZMnGGtzkLSaWc8OvY8QfPsdmAR2j+slI=";
+  };
+
+  frontend = buildNpmPackage {
+    pname = "fusion-frontend";
+    inherit version;
+    src = "${src}/frontend";
+
+    patches = [ ./frontend-pin-version.patch ];
+
+    npmDepsHash = "sha256-sOdviGGVB41IjO2tpw655dSxuqxufQGo2CqmsjNCDn0=";
+
+    installPhase = ''
+      runHook preInstall
+
+      cp -r build $out
+
+      runHook postInstall
+    '';
+  };
+
+  vendorHash = "sha256-isSoDDJLWIxmihu9txcPMDBJ+l323yrfXqyhqtEAoUI=";
+
+  subPackages = [ "cmd/server" ];
+
+  overrideModAttrs = prev: {
+    nativeBuildInputs = prev.nativeBuildInputs ++ [ mockgen ];
+
+    preBuild = ''
+      go generate ./...
```

Comment:
As of 0.8.11, this is no longer needed (see https://github.com/0x2E/fusion/commit/d6194e3fc72d3375b286500f33402e6e752d7c72)